drivers/sensor/current_amp/current_amp.c

```diff
@@ -8,27 +8,36 @@
 
 #include <zephyr/drivers/adc.h>
 #include <zephyr/drivers/adc/current_sense_amplifier.h>
+#include <zephyr/drivers/gpio.h>
 #include <zephyr/drivers/sensor.h>
+#include <zephyr/pm/device.h>
 
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(current_amp, CONFIG_SENSOR_LOG_LEVEL);
 
+struct current_amp_config {
+	struct current_sense_amplifier_dt_spec current_sense;
+#ifdef CONFIG_PM_DEVICE
+	struct gpio_dt_spec gpio_power;
+#endif
+};
```

Comment:
how about having the gpio part of `struct current_sense_amplifier_dt_spec` instead? It is defined to match the binding so I think it would make more sense that way. One extra pointer all the time if you don't need it but then it's in flash so prob not a big deal.

Reply:
Done 👍